shell/plugins/internal-api/shared/base-api.ts

```diff
@@ -0,0 +1,11 @@
+export abstract class BaseApi {
+  // The Vuex store, available to all API classes
+  protected $store: any;
```

Comment:
Do you think we'll need an app instance? Or do you want the derived classes to take care of that?